test/quantization/test_quantize_fx.py

```diff
@@ -3092,28 +3094,145 @@ def test_instance_norm(self):
                 quantized_module, torch.ops.quantized.instance_norm,
                 skip_op_arg_for_functional=True)
 
-    def test_silu(self):
+    def _test_default_node_quant_handler_ops(self, module, functional, qconfig, is_reference=True, node_list=[], additional_quant_pattern_dict={}):
         class M(torch.nn.Module):
-            def __init__(self):
+            def __init__(self, mod, func):
                 super().__init__()
-                self.silu = torch.nn.SiLU()
+                self.module = mod()
+                self.functional = func
 
             def forward(self, x):
-                x = self.silu(x)
-                x = torch.nn.functional.silu(x)
+                x = self.module(x)
+                x = self.functional(x)
                 return x
 
-        data = (torch.randn((2, 2, 2, 2), dtype=torch.float),)
+        data = torch.randn((2, 2, 2, 2))
         quant_type = QuantType.STATIC
-        qconfig_dict = {
-            "": float16_static_qconfig
-        }
-        node_occurrence = {
-            ns.call_method("to"): 3
-        }
-        m = self.checkGraphModeFxOp(
-            M(), data, quant_type, custom_qconfig_dict=qconfig_dict,
-            expected_node_occurrence=node_occurrence)
+        custom_qconfig_dict = {"additional_quant_pattern": additional_quant_pattern_dict}
+        qconfig_dict = {"": qconfig}
+
+        m = M(module, functional).eval()
+        m_prep = torch.quantization.quantize_fx.prepare_fx(m, qconfig_dict, custom_qconfig_dict)
+        m_prep(data)
+        m_quant = torch.quantization.quantize_fx.convert_fx(m_prep, is_reference=is_reference)
+        m_quant(data)
+
+        self.checkGraphModuleNodes(m_quant, expected_node_list=node_list)
+
+    def test_gelu_normal(self):
+        module = torch.nn.GELU
+        functional = torch.nn.functional.gelu
+        qconfig = torch.quantization.get_default_qconfig("fbgemm")
+        # qconfig = {"object_type": [(module, None), (functional, qconfig)]}
+        is_reference = False
+        node_list = [
+            ns.call_module(module),
+            ns.call_function(functional),
+        ]
+        self._test_default_node_quant_handler_ops(
+            module, functional, qconfig, is_reference, node_list)
+
+    def test_softmax_normal(self):
+        module = torch.nn.Softmax
+        functional = torch.nn.functional.softmax
+        qconfig = torch.quantization.get_default_qconfig("fbgemm")
+        is_reference = False
+        node_list = [
+            ns.call_module(module),
+            ns.call_function(functional),
+        ]
+        self._test_default_node_quant_handler_ops(
+            module, functional, qconfig, is_reference, node_list)
+
+    def test_gelu_reference(self):
```

Comment:
Can we combine the reference tests for gelu, softmax and silu into a single one?

Reply:
Got it, feels like an acceptable tradeoff: larger test files, but cleaner test signals.